src/components/switch/switch.tsx

```diff
@@ -0,0 +1,37 @@
+import { BaseComponent } from "../common/base-component";
+import { Component, Event, EventEmitter, Prop } from "@stencil/core";
+import { SwitchRender } from "../renders/bootstrap/switch/switch-render";
+@Component({
+  host: {
+    role: "switch"
+  },
+  shadow: false,
+  styleUrl: "switch.scss",
+  tag: "gx-switch"
+})
+export class Switch extends SwitchRender(BaseComponent) {
+  /**
+   * Attribute that provides the caption to the <label> tag.
```

Comment:
I woudln't mention the `<label>` here. The usage of a `<label>` element is an implementation detail that should be hidden to the component end user.